File: web/src/services/getProjectById.ts
import { callApi } from "./api";
import { Service } from "./getServicesByDeployId";

export type AppDto = {
  id: string;
  path: string;
  name: string;
  isDockerFile: boolean;
  isDockerCompose: boolean;
};

export type ProjectDto = {
  id: string;
  name: string;
  path: string;
  serverId: string;
  services: Service[];
};

export async function getProjectByIdApi(id: string): Promise<ProjectDto> {
  return await callApi<ProjectDto>("/project/" + id, "GET");
}

File: internal/application/get-project-by-id.go
package application

import (
	"cchalop1.com/deploy/internal/api/graph/model"
	"cchalop1.com/deploy/internal/api/service"
)

func GetProjectById(deployService *service.DeployService, id string) (*model.Project, error) {
	product, err := deployService.DatabaseAdapter.GetProjectById(id)

	if err != nil {
		return nil, err
	}

	return &model.Project{
		ID:       product.Id,
		Name:     product.Name,
		Path:     product.Path,
		Services: []*model.Service{},
		Apps:     []*model.App{},
	}, nil
}
